actions/task.go

```diff
@@ -0,0 +1,303 @@
+		return c.Error(http.StatusNotFound, err)
+	}
+
+	if err := tx.Destroy(tasks); err != nil {
+		return err
+	}
+
+	return c.Redirect(302, "/table-incomplete")
+}
+func ShowEdit(c buffalo.Context) error {
+    
+	tasks := &models.Task{}
+	
+	tx, ok := c.Value("tx").(*pop.Connection)
+	if !ok {
+		return errors.New("no transaction found")
+	}
+
+	q := tx.Where("finished_at is null")
+	count, err := q.Count(tasks)
+	if err != nil {
+		return err
+	}
+
+	countInfoIncomplete := fmt.Sprintf("%v Incomplete Tasks", count)
+
+	c.Set("count", countInfoIncomplete)
+
+	if err := tx.Find(tasks, c.Param("task_id")); err != nil {
+		return c.Error(http.StatusNotFound, err)
+	}
+
+	c.Set("tasks", tasks)
+
+	return c.Render(http.StatusOK, r.HTML("edit_task.plush.html"))
+}
+
+func Update(c buffalo.Context) error {
+	
+	tasks := models.Task{}
+
+	tx, ok := c.Value("tx").(*pop.Connection)
+	if !ok {
+		return errors.New("no transaction found")
+	}
+
+	if err := tx.Find(&tasks, c.Param("task_id")); err != nil {
+		return c.Error(http.StatusNotFound, err)
+	}
+
+	if err := c.Bind(&tasks); err != nil {
+		return err
+	}
+	q := tx.Where("finished_at is null")
+	count, err := q.Count(tasks)
+	if err != nil {
+		return err
+
+	}
+
+	countInfoIncomplete := fmt.Sprintf("%v Incomplete Tasks", count)
+
+	c.Set("count", countInfoIncomplete)
+	c.Set("tasks", &tasks)
+
+	verrs, err := tx.ValidateAndUpdate(&tasks)
+	if err != nil {
+		return err
+	}
+	if verrs.HasAny() {
+		c.Set("error", verrs.Get("name"))
+		return c.Render(422, r.HTML("edit_task.plush.html"))
+	}
+
+	// if tasks.Name_task == "" {
+	// 	c.Flash().Add("danger", "Alert enter task name!")
+	// } else {
+	// 	if err := tx.Update(tasks); err != nil {
+	// 		return c.Error(http.StatusNotFound, err)
+	// 	}
+	// }
+
+	return c.Redirect(302, "/table-incomplete")
+}
+
+func Check(c buffalo.Context) error {
+	
+	tasks := &models.Task{}
+
+	tx, ok := c.Value("tx").(*pop.Connection)
+	if !ok {
+		return errors.New("no transaction")
+	}
+
+	if err := tx.Find(tasks, c.Param("task_id")); err != nil {
+		return c.Error(http.StatusNotFound, err)
+	}
+
+	if !tasks.FinishedAt.Valid {
+		tasks.FinishedAt = nulls.NewTime(time.Now())
+		if err := tx.Update(tasks); err != nil {
+			return c.Error(http.StatusNotFound, err)
+		}
+		return c.Redirect(302, "/table-incomplete")
+	}
+
+	c.Flash().Add("danger text-center", "This task is already complete!")
+	return c.Redirect(302, "/table-incomplete")
+}
+
+func UnCheck(c buffalo.Context) error {
+
+	tasks := &models.Task{}
+
+	tx, ok := c.Value("tx").(*pop.Connection)
+	if !ok {
+		return errors.New("no transaction")
+	}
+
+	
+
+	if err := tx.Find(tasks, c.Param("task_id")); err != nil {
+		return c.Error(http.StatusNotFound, err)
+	}
+
+	if tasks.FinishedAt.Valid {
+		tasks.FinishedAt = nulls.Time{}
+
+		if err := tx.Update(tasks); err != nil {
+			return c.Error(http.StatusNotFound, err)
+		}
+		return c.Redirect(302, "/table-incomplete")
```

Comment:
Please to use the constants of the package "http" to set the http status codes when ever it is needed.